openhcl/virt_mshv_vtl/src/processor/tdx/mod.rs

```diff
@@ -1842,12 +1914,23 @@ impl UhProcessor<'_, TdxBacked> {
                         cpl: exit_info.cpl(),
                     })
                 {
-                    self.runner
-                        .set_vp_register(intercepted_vtl, HvX64RegisterName::Xfem, value.into())
-                        .map_err(|err| {
-                            VpHaltReason::Hypervisor(UhRunVpError::EmulationState(err))
-                        })?;
-                    self.advance_to_next_instruction(intercepted_vtl);
+                    if self.cvm_protect_secure_register_write(
+                        intercepted_vtl,
+                        HvX64RegisterName::Xfem,
+                        value,
+                    ) {
+                        // Once the intercept message has been posted, no further
```

Comment:
I'm not loving this copy/paste comment block everywhere. Can we rename the function to something like `try_intercept_secure_register_write`, such that it should be more obvious to the caller that if it returns false, we need to handle the instruction?